README.md

```diff
@@ -213,4 +213,42 @@ if err != nil {
     fmt.Println(err)
 }
 fmt.Println(keys)
-```
\ No newline at end of file
+```
+
+### Fetching List Key In Sorted Ascending Order Based On Paramaeters.
+
+```go
+var sortObj SortBy
+sortStr := SortBy(sortObj)
+sortStr = sortStr.WithID()
+sortStr = sortStr.WithCretionDate()
+
+listKeysOptions := &kp.ListKeysOptions{
+  Sort:sort,
+}
+
+keys, err := client.ListKeys(ctx, listKeysOptions)
+if err != nil {
+    fmt.Println(err)
+}
+fmt.Println(keys)
+```
+
+### Fetching List Key In Sorted Descending Order Based On Paramaeters.
+
+```go
+var sortObj SortBy
+sortStr := SortBy(sortObj)
```

Comment:
Why did you make it an extended type vs doing a composition/encapsulation?

i.e. why not
```
type SortBy struct {
    sortStr string[]
}

...

mysort := SortBy{}.WithCreationDateDesc();

res := ListKeys(...., ListKeysOptions{sortby: mySort});

```

Reply:
I changed the design pattern to handle it differently, let me know does it achieve what you are suggesting